### ISSUE TITLE
Support for Unicode characters in home folder names during installation on Windows

### ISSUE THREAD

**Issue by coditect** [ladybug-tools/lbt-grasshopper#1002]
The LBSetup component in `installer.gh` fails when it is run on Windows by a user whose home folder contains certain Unicode characters.  Symptoms are [described on Discourse](https://discourse.ladybug.tools/t/installation-issues-with-non-ascii-characters/11274).

This issue is a result of [buggy Unicode support in the subprocess module](https://bugs.python.org/issue1759845) in Python 2.7.  The official solution is to upgrade to Python 3, but that is unfortunately not an option with IronPython.  However, IronPython does give us an opportunity to work around the problems with the subprocess module by invoking `pip` through the .NET API instead.  The code below uses the .NET `Process` object to execute a shell command expressed as a list of strings that would normally be passed to `subprocess.Popen`:

```python
if os.name == 'nt':
    from System.Diagnostics import Process, ProcessStartInfo

class EventDataRecorder(list):

    def __call__(self, sender, event):
        if event.Data is not None:
            self.append(event.Data)
    
    def __str__(self):
        return "\n".join(self)

def run_subprocess_via_dotnet(cmds):
    try:
        start_info = ProcessStartInfo(
            RedirectStandardOutput = True,
            RedirectStandardError = True,
            UseShellExecute = False,
            CreateNoWindow = True,
            FileName = cmds[0],
            Arguments = " ".join(cmds[1:])
        )
        output = EventDataRecorder()
        errors = EventDataRecorder()
        process = Process(StartInfo=start_info)
        process.OutputDataReceived += output
        process.ErrorDataReceived += errors
        process.Start()
        process.BeginOutputReadLine()
        process.BeginErrorReadLine()
        process.WaitForExit()
        return str(output), str(errors)
    except WindowsError as e:
        return '', e.strerror
    except Exception as e:
        return '', str(e)
```

The `update_libraries_pip` function could be modified to use the .NET API instead of `subprocess.Popen` when required:
```python
if os.name == 'nt' and python_exe.encode('mbcs') != python_exe:
    output = run_subprocess_via_dotnet(cmds)
else:
    use_shell = True if os.name == 'nt' else False
    process = subprocess.Popen(cmds, shell=use_shell, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
    output = process.communicate()

stdout, stderr = output
return stderr
```

**Comment by coditect** [comment]
It is important to note the code above is not sufficient to allow users with Unicode home folder names to install Ladybug.  There is also [an issue](https://github.com/ladybug-tools/lbt-grasshopper/files/6290560/lbt_install_output.txt) with installing `luigi` (a dependency of `lbt-recipies`) that would need to be worked around.  It is not clear to me right now whether this is a problem with `pip`, the compilation of `luigi`, or something else.

**Comment by chriswmackey** [comment]
Yes, and those issues are just the tip of the iceberg since such users will experience even more issues even after then install the software. See here for an explanation:
https://github.com/ladybug-tools/lbt-grasshopper/issues/1001

**Comment by chriswmackey** [comment]
This has all been addressed in LBT 1.5 and above. As long as the user installs with [the Pollination Single-Click installer available here](https://app.pollination.cloud/cad-plugins), they will be able to use the plugin while having non-unicode characters in their username.
